examples/dispatching2_features.rs

```diff
@@ -0,0 +1,162 @@
+//! This example provide quick overview of the new features in the
+//! `dispatching2` module.
+
+use rand::Rng;
+/// Note that you need to import `prelude2` because `prelude` contains
+/// items from the old dispatching system (it may change in future versions).
+use teloxide::prelude2::*;
+use teloxide::{types::Update, utils::command::BotCommand};
+
+#[tokio::main]
+async fn main() {
+    // Start you main as early: start logging, create bot, etc.
+    teloxide::enable_logging!();
+    log::info!("Starting dispatching2_features_bot...");
+    let bot = Bot::from_env().auto_send();
+
+    let parameters = ConfigParameters {
+        bot_maintainer: 268486177, // Paste your ID if you run this bot.
+        maintainer_username: None,
+    };
+
+    let handler = Update::filter_message()
+        // Branch is a special method that allow you to handle update several ways.
+        .branch(
+            // Filter allow you to filter updates by some condition.
+            dptree::filter(
+                // Note that `async move` is obligatory.
+                |msg: Message| async move { msg.chat.is_group() || msg.chat.is_supergroup() },
+            )
+            // Endpoint is a last message handler.
+            .endpoint(|msg: Message, bot: AutoSend<Bot>| async move {
+                log::info!("Received message from the group chat.");
+                bot.send_message(msg.chat.id, "This is a group chat.").await?;
+                respond(())
+            }),
+        )
+        // Note that we cannot filter messages from public chats in the next branch,
+        // because they all is handled by previous branch.
+        .branch(
+            // There are some `filter` functions on message, that filters events. This
+            // filter will filter only messages with dices.
+            Message::filter_dice().endpoint(|msg: Message, bot: AutoSend<Bot>| async move {
+                bot.send_message(msg.chat.id, "This is a dice!")
+                    .reply_to_message_id(msg.id)
+                    .await?;
+                Ok(())
+            }),
+        )
+        .branch(
+            // If you do not like photos, you can break their handling like that.
+            Message::filter_photo().endpoint(|| async move { Ok(()) }),
+        )
+        .branch(
+            dptree::entry()
+                // This method allows to parse text messages commands.
+                .add_command::<SimpleCommand>()
```

Comment:
Renamed `add_command` => `filter_command`.

Reply:
> Does it need to be an extension of an entry? Can't we just make command_entry::<SimpleCommand>()?

`.add_command` can be _chained_ with other handlers. `command_entry::<SimpleCommand>()` can be chained too with smth like `.branch(...)` but it would be confusing. Maybe we should think about it in the next release.

Reply:
I mean most of the time (on of) the first thing(s) you want is to parse the command, so `command_entry` can probably be used 90% of the time. (to be clear: I propose to later add `command_entry` without removing `filter_command`).

Unresolving so we don't forget about this, while preparing next release.